packages/client/src/pages/Game/maps/Map.ts

```diff
@@ -1,83 +1,245 @@
   ) {
+    // set tile center
     this.mapParams.tileCenter = this.mapParams.gridStep / 2;
-    this.mapParams.rightBorder = this.mapParams.width / 2;
+
+    // set map start and end stages
+    this.stageArr = [
+      { direction: "start", limit: { x: 0, y: this.tileToNumber(1) } },
+      {
+        direction: "right",
+        limit: { x: this.tileToNumber(14), y: this.tileToNumber(1) },
+      },
+      {
+        direction: "down",
+        limit: { x: this.tileToNumber(14), y: this.tileToNumber(4) },
+      },
+      {
+        direction: "left",
+        limit: { x: this.tileToNumber(1), y: this.tileToNumber(4) },
+      },
+      {
+        direction: "down",
+        limit: { x: this.tileToNumber(1), y: this.tileToNumber(6) },
+      },
+      {
+        direction: "right",
+        limit: { x: this.tileToNumber(14), y: this.tileToNumber(6) },
+      },
+      {
+        direction: "down",
+        limit: { x: this.tileToNumber(14), y: this.tileToNumber(9) },
+      },
+      {
+        direction: "left",
+        limit: { x: this.tileToNumber(1), y: this.tileToNumber(9) },
+      },
+      {
+        direction: "down",
+        limit: { x: this.tileToNumber(1), y: this.tileToNumber(14) },
+      },
+      {
+        direction: "right",
+        limit: { x: this.tileToNumber(5), y: this.tileToNumber(14) },
+      },
+      {
+        direction: "up",
+        limit: { x: this.tileToNumber(5), y: this.tileToNumber(11) },
+      },
+      {
+        direction: "right",
+        limit: { x: this.tileToNumber(8), y: this.tileToNumber(11) },
+      },
+      {
+        direction: "down",
+        limit: { x: this.tileToNumber(8), y: this.tileToNumber(14) },
+      },
+      {
+        direction: "right",
+        limit: { x: this.tileToNumber(10), y: this.tileToNumber(14) },
+      },
+      {
+        direction: "up",
+        limit: { x: this.tileToNumber(10), y: this.tileToNumber(11) },
+      },
+      {
+        direction: "right",
+        limit: { x: this.tileToNumber(12), y: this.tileToNumber(11) },
+      },
+      {
+        direction: "down",
+        limit: { x: this.tileToNumber(12), y: this.tileToNumber(14) },
+      },
+
+      {
+        direction: "right",
+        limit: { x: this.tileToNumber(14), y: this.tileToNumber(14) },
+      },
+      {
+        direction: "up",
+        limit: { x: this.tileToNumber(14), y: this.tileToNumber(12) },
+      },
+      {
+        direction: "right",
+        limit: { x: this.tileToNumber(15), y: this.tileToNumber(12) },
+      },
+      {
+        direction: "end",
+        limit: { x: this.tileToNumber(16), y: this.tileToNumber(12) },
+      },
+    ];
 
     // create mapTilesArr
     this.createMapTilesArr();
 
     // pop tiles which is occupied by map path
     this.popMapPathTiles();
+
+    this.renderParams.mapRoadSpriteSource = new Image(256, 448);
+    this.renderParams.mapRoadSpriteSource.src =
+      "/sprites/map/mapRoadSprite.png";
```

Comment:
don't use hardcoded paths. In bundle they will not change and break.

Import picture at the top of the file and use it's value as path:
```suggestion

import mapRoadSprite from "/sprites/map/mapRoadSprite.png";
// ...
    this.renderParams.mapRoadSpriteSource.src = mapRoadSprite
      
```

Reply:
this ll be reworked in next minor release, it's here just to proof-of-concept